app/src/main/java/com/danielkim/soundrecorder/RecordingService.java

```diff
@@ -121,9 +121,15 @@ public void setFileNameAndPath(){
     }
 
     public void stopRecording() {
-        mRecorder.stop();
-        mElapsedMillis = (System.currentTimeMillis() - mStartingTimeMillis);
-        mRecorder.release();
+        try {
+            mRecorder.stop();
+            mElapsedMillis = (System.currentTimeMillis() - mStartingTimeMillis);
+            mRecorder.release();
+
+        } catch (RuntimeException e) {
+            Log.e(LOG_TAG,"Runtime Exception");
```

Comment:
What is the actual cause of this exception?

Reply:
MediaRecorder object is still in prepare() state and release() is called.

Reply:
Could you log `e.getMessage()`?

Reply:
I mean include the message in the source:

```java
Log.e(LOG_TAG, "Runtime Exception: " + e.getMessage());
```

Reply:
Also, if there is indeed this exception, wouldn't the next line (Toast.makeText(this, getString(R.string.toast_recording_finish) + " " + mFilePath, Toast.LENGTH_LONG).show();) not be applicable because nothing has actually be recorded?